src/main/java/com/traveltime/sdk/dto/requests/TimeFilterFastProtoRequest.java

```diff
@@ -85,11 +87,32 @@ private byte[] createByteArray() {
                 .setLng(this.originCoordinate.getLng().floatValue())
                 .build();
 
-        RequestsCommon.Transportation transportation = RequestsCommon
+        RequestsCommon.Transportation.Builder transportationBuilder = RequestsCommon
                 .Transportation
                 .newBuilder()
-                .setTypeValue(this.transportation.getCode())
-                .build();
+                .setTypeValue(this.transportation.getType().getCode());
+
+        if (this.transportation.getDetails() instanceof PublicTransportDetails) {
```

Comment:
Instead of this kind of matching, I think it would be better to implement a visitor/fold pattern in `TransportDetails`, that is 
```
interface TransportDetails {
   <T> fold(Function<PublicTransportDetails, T> ptFn, Function<DrivingAndPublicTransportDetails, T> drPtFn);
}
```

Reply:
Or we could strongly couple it to the proto implementation (shouldn't be a problem for SDK code IMO)
```
interface TransportDetails {
   void setDetails(RequestsCommon.Transportation.Builder builder);
}
```